synapse_data_warehouse/database_roles/V2.33.0__database_analyst_roles.sql

```diff
@@ -0,0 +1,96 @@
+USE DATABASE {{ database_name }}; --noqa: JJ01,PRS,TMP
+
+-- Create roles which will aggregate the object-type-specific roles 
+CREATE OR REPLACE DATABASE ROLE SYNAPSE_ALL_ANALYST;
+CREATE OR REPLACE DATABASE ROLE SYNAPSE_RAW_ALL_ANALYST;
+CREATE OR REPLACE DATABASE ROLE SCHEMACHANGE_ALL_ANALYST;
+
+-- Grant ownership of the analyst aggregate roles to their
+-- respective admin aggregate role
+GRANT OWNERSHIP
+    ON DATABASE ROLE SYNAPSE_ALL_ANALYST
+    TO ROLE SYNAPSE_ALL_ADMIN; --noqa: JJ01,PRS,TMP
+GRANT OWNERSHIP
+    ON DATABASE ROLE SYNAPSE_RAW_ALL_ANALYST
+    TO ROLE SYNAPSE_RAW_ALL_ADMIN; --noqa: JJ01,PRS,TMP
+GRANT OWNERSHIP
+    ON DATABASE ROLE SCHEMACHANGE_ALL_ANALYST
+    TO ROLE SCHEMACHANGE_ALL_ADMIN; --noqa: JJ01,PRS,TMP
+
+-- Grant analyst aggregate roles to the database analyst account role
+GRANT DATABASE ROLE SYNAPSE_ALL_ANALYST
+    TO ROLE {{ database_name }}_ANALYST; --noqa: JJ01,PRS,TMP
+GRANT DATABASE ROLE SYNAPSE_RAW_ALL_ANALYST
+    TO ROLE {{ database_name }}_ANALYST; --noqa: JJ01,PRS,TMP
+GRANT DATABASE ROLE SCHEMACHANGE_ALL_ANALYST
+    TO ROLE {{ database_name }}_ANALYST; --noqa: JJ01,PRS,TMP
+
+
+-- Create object-type-specific roles, grant ownership of each to
+-- their respective admin aggregate role, and grant each to the
+-- respective analyst aggregate role
+-- SYNAPSE
+CREATE OR REPLACE DATABASE ROLE SYNAPSE_TABLE_READ;
+CREATE OR REPLACE DATABASE ROLE SYNAPSE_STAGE_READ;
+CREATE OR REPLACE DATABASE ROLE SYNAPSE_VIEW_READ;
+CREATE OR REPLACE DATABASE ROLE SYNAPSE_TASK_READ;
+
+GRANT OWNERSHIP
+	ON DATABASE ROLE SYNAPSE_TABLE_READ
+	TO DATABASE ROLE SYNAPSE_ALL_ADMIN;
+GRANT OWNERSHIP
+	ON DATABASE ROLE SYNAPSE_STAGE_READ
+	TO DATABASE ROLE SYNAPSE_ALL_ADMIN;
+GRANT OWNERSHIP
+	ON DATABASE ROLE SYNAPSE_VIEW_READ
+	TO DATABASE ROLE SYNAPSE_ALL_ADMIN;
+GRANT OWNERSHIP
+	ON DATABASE ROLE SYNAPSE_TASK_READ
+	TO DATABASE ROLE SYNAPSE_ALL_ADMIN;
+
+GRANT DATABASE ROLE SYNAPSE_TABLE_READ
+	TO DATABASE ROLE SYNAPSE_ALL_ANALYST;
+GRANT DATABASE ROLE SYNAPSE_STAGE_READ
+	TO DATABASE ROLE SYNAPSE_ALL_ANALYST;
+GRANT DATABASE ROLE SYNAPSE_VIEW_READ
+	TO DATABASE ROLE SYNAPSE_ALL_ANALYST;
+GRANT DATABASE ROLE SYNAPSE_TASK_READ
+	TO DATABASE ROLE SYNAPSE_ALL_ANALYST;
+
+-- SYNAPSE_RAW
+CREATE OR REPLACE DATABASE ROLE SYNAPSE_RAW_TABLE_READ;
+CREATE OR REPLACE DATABASE ROLE SYNAPSE_RAW_STAGE_READ;
+CREATE OR REPLACE DATABASE ROLE SYNAPSE_RAW_STREAM_READ;
+CREATE OR REPLACE DATABASE ROLE SYNAPSE_RAW_TASK_READ;
```

Comment:
STAGE should be swapped with STREAM for the SYNAPSE_RAW object-specific database roles in the following diagrams. 
![SNOW-200 (8)](https://github.com/user-attachments/assets/06cf1de5-3119-4297-9867-350130ddf013)